pkg/controller/scyllacluster/resource.go

```diff
@@ -648,17 +650,47 @@ func stringOrDefault(str, def string) string {
 	return def
 }
 
-func MakeServiceAccount(sc *scyllav1.ScyllaCluster) *corev1.ServiceAccount {
+func MakeServiceAccount(sc *scyllav1.ScyllaCluster, serviceAccounts map[string]*corev1.ServiceAccount, serviceAccountLister corev1listers.ServiceAccountLister) (*corev1.ServiceAccount, error) {
+	var annotations map[string]string
+	name := naming.MemberServiceAccountNameForScyllaCluster(sc.Name)
+
+	// In 1.7.0 we started managing ServiceAccounts. Because previously these were created by users
+	// there's a high chance that they don't match our selector, hence are not available in serviceAccounts.
+	//
+	// We need to copy annotations stored in user SA because these may control integrations with k8s provider services
+	// like IAM management.
+	// Cache fetch happens only once - when SA is being adopted. It can be removed in 1.8.0 because all SAs will be already
+	// adopted, and they will match our selector.
+	existing, exists := serviceAccounts[name]
+	if !exists {
+		var err error
+		existing, err = serviceAccountLister.ServiceAccounts(sc.Namespace).Get(name)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("can't get %s service account: %w", naming.ManualRef(sc.Namespace, name), err)
+		}
+		exists = !apierrors.IsNotFound(err)
+	}
+
+	if exists {
+		if existing.Annotations != nil {
+			annotations = make(map[string]string)
+		}
```

Comment:
nit: Although this is technically correct, would it be clearer to just always create a non nil map at the beginning?

Reply:
There's a difference between empty map and nil map, although it doesn't matter in this case. Added initialization to have shorter code.